### PR TITLE
Tagging netgo to be used while compiling with cgo 

### PR DESCRIPTION
**Pull request by olljanat**
~Implements suggestion in https://github.com/longhorn/longhorn/issues/4415#issuecomment-1214885963 and disable CGO here too like it was done in longhorn-manager by https://github.com/longhorn/longhorn-manager/pull/61~

Looks that instance-manager needs CGO so switched to alternative solution proposed in https://github.com/longhorn/longhorn/issues/1768#issuecomment-690778513

### CODE REVIEW

File: scripts/build [hunk 1]
```diff
@@ -6,7 +6,7 @@ source $(dirname $0)/version
 cd $(dirname $0)/..
 
 mkdir -p bin
-go build -ldflags \
+go build --tags netgo -ldflags \
```
Comment (innobead) [inline scripts/build:9]:
```suggestion
go build -tags netgo -ldflags \
```

**Review by innobead — CHANGES_REQUESTED** [review]

File: scripts/build (hunk 1 above)
Comment (olljanat) [reply scripts/build]: Good catch. Fixed.

**Comment by olljanat** [comment]
@innobead typo fixed, please check again.

**Comment by olljanat** [comment]
PR is still valid

**Comment by innobead** [comment]
Because the instance manager has the proxy capability and it is actually the core of the data plane, we need this as well instead of longhorn-engine only. (probably we will reach out to the service in DNS from the instance manager)

**Review by innobead — APPROVED** [review]
Merge after QA verification. 

cc @longhorn/qa @roger-ryao

**Comment by roger-ryao** [comment]
> Merge after QA verification.
> 
> cc @longhorn/qa @roger-ryao

Hi @innobead 
- [core test report](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2448/)
- [regression test report](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2447/)
- [The failed cases re-run report](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2457/)

The test PASS, we could merge longhorn-instance-manager PR
cc @longhorn/qa

**Pull request merged**